packages/lime-system/files/usr/lib/lua/lime/utils.lua

```diff
@@ -543,4 +543,20 @@ function utils.is_valid_mac(string)
     end
 end
 
+function utils.deepcompare(t1,t2)
```

Comment:
As this implementation is recursive, it might have a heap issue.
You can explore an iterative one, that is cheaper and equally fast.

Reply:
would also be good to have tests for these implementations, both for documenting their use and for robustness

Reply:
I added tests for this function, but couldn't come up with a non recursive implementation quickly.
I understand you mean stack issue instead of heap.

Reply:
An early check `if ty1 == ty2 then return true end` would optimize a bunch of cases ;)
Moreover this implementation would overflow the heap in case of a back-reference inside the table

Reply:
German already addressed this issues. In my opinion is ok to have a recursive strategy.